### ISSUE TITLE
WAR model is full of hacks

### ISSUE THREAD

**Issue by ccbrown** [ccbrown/xiv-sim#46]
Several abilities need to be more properly implemented.

**Comment by jingai** [comment]
Just curious, but what exactly?

**Comment by ccbrown** [comment]
The things you commented.

https://github.com/ccbrown/xiv-sim/blob/master/src/models/Warrior.cpp#L17
https://github.com/ccbrown/xiv-sim/blob/master/src/models/Warrior.cpp#L209
https://github.com/ccbrown/xiv-sim/blob/master/src/models/Warrior.cpp#L264

**Comment by jingai** [comment]
Yeah, it wasn't immediately obvious how I could drop the debuff for just the current action.  I'm sure you can figure a way though.
